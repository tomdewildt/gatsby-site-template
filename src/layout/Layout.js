import React from "react";
import { ThemeProvider } from "styled-components";
import PropTypes from "prop-types";

import theme, { GlobalStyle } from "../theme";

const Layout = ( { children } ) => (
    <ThemeProvider theme={ theme }>
        <GlobalStyle />
        { children }
    </ThemeProvider>
);

Layout.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Layout;
