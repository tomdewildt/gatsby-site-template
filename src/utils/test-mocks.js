jest.mock( "gatsby", () => ( {
    ...jest.requireActual( "gatsby" ),
    graphql: jest.fn( () => "" ),
    useStaticQuery: jest.fn( () => ( {
        site: {
            siteMetadata: {
                siteTitle: "title",
                siteLanguage: "en",
                siteDescription: "description",
                siteBanner: "https://example.com/banner.png",
                siteUrl: "https://example.com",
            },
        },
    } ) ),
} ) );
