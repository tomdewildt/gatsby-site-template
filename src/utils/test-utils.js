/* eslint-disable import/no-extraneous-dependencies,react/prop-types */
import React from "react";
import { ThemeProvider } from "styled-components";
import { render, waitFor } from "@testing-library/react";

import defaultTheme from "../theme";

export const findByAttribute = async ( key, value ) => {
    let result = null;
    await waitFor( () => {
        result = document.querySelector( `[${ key }='${ value }']` );
        return expect( result ).toBeInTheDocument();
    } );
    return result;
};

export const findAllByAttribute = async ( key, value ) => {
    let result = null;
    await waitFor( () => {
        result = document.querySelectorAll( `[${ key }='${ value }']` );
        return expect( result.length ).toBeGreaterThan( 0 );
    } );
    return result;
};

export const renderWithTheme = (
    component,
    theme = defaultTheme,
) => ( {
    findByAttribute,
    findAllByAttribute,
    ...render( component, {
        wrapper: ( { children } ) => (
            <ThemeProvider theme={ theme }>
                { children }
            </ThemeProvider>
        ),
    } ),
} );

export const renderComponent = (
    component,
    options = {},
) => ( {
    findByAttribute,
    findAllByAttribute,
    ...render( component, options ),
} );
