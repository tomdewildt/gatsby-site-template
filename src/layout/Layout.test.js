import React from "react";
import { axe } from "jest-axe";

import { renderComponent } from "../utils/test-utils";

import Layout from "./Layout";

describe( "Layout", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderComponent( <Layout>layout</Layout> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderComponent( <Layout>layout</Layout> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );
} );
