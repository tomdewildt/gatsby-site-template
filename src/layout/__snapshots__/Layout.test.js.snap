// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders consistently 1`] = `
<DocumentFragment>
  layout
</DocumentFragment>
`;
